src/main/java/frc/robot/commands/internals/ShootCommand.java

```diff
@@ -18,11 +20,14 @@ public ShootCommand(InternalSubsystem internals) {
 
     @Override
     public void initialize() {
-        internals.requestShot(() -> finished = true);
+        internals.requestShot();
+        shotRequested = true;
     }
 
     @Override
     public boolean isFinished() {
+        boolean doneShooting = shotRequested && !internals.shotRequested();
+        this.finished = finished || doneShooting; // in case someone ever queues a new shot before successive isFinished calls (???)
```

Comment:
I don't think it's realistic for a driver to request a new shot in the 20ms periodic window between the first tick after the old shot is fired and the tick immediately following it. Also, drivers won't be able to request shots in autonomous (which is when this command would be used) and there's no way for two calls to this command in a SequentialCommandGroup to trigger this because the first command's `isFinished()` needs to return true before the second command's `initialize()` runs.

For `isFinished()`, we may not have to store `shotRequested` either; if `initialize()` always runs before the first call to `isFinished()`, `isFinished()` can be implemented like
```java
@Override
public boolean isFinished() {
    return !internals.shotRequested();
}
```

Reply:
ok, does it hurt?

Reply:
I guess not? But it's an impossible scenario anyways and it adds another field variable